lib/nrf/common/i2c_common.c

```diff
@@ -52,14 +55,15 @@ void i2c_disable(uint32_t i2c)
 }
 
 /** @brief Start I2C transmission.
+ *
+ * Starts STARTTX task, which generates start condition on I2C bus and
+ * transmits address previously configured by @ref i2c_set_address.
  *
  * @param[in] i2c uint32_t i2c peripheral base.
- * @param[in] data uint8_t the first byte to send.
  */
-void i2c_start_tx(uint32_t i2c, uint8_t data)
+void i2c_start_tx(uint32_t i2c)
 {
-	PERIPH_TRIGGER_TASK(I2C_TASK_STARTTX(i2c));
-	I2C_TXD(i2c) = data;
+	I2C_TASK_STARTTX(i2c) = 1;
```

Comment:
what was the "PERIPH_TRIGER_TASK() " wrapper doing?

Reply:
never mind, I see it was a stupid dumb macro that should have been burnt in fire.